pandas/core/series.py

```diff
@@ -4242,6 +4242,9 @@ def _try_cast(arr, take_fast_path):
                                            not (is_iterator(subarr) or
                                            isinstance(subarr, np.ndarray))):
                 subarr = construct_1d_object_array_from_listlike(subarr)
+            elif type(subarr).__name__ == "DatetimeArray":
```

Comment:
woa??

Reply:
Yah this is a kludge that should have gotten changed to an `isinstance(subarr, ABCDatetimeArray)` before pushing